Data/Config.hs

```diff
@@ -159,6 +161,49 @@ loadConfig path
              Left e    -> throwM e
              Right reg -> return $ Config reg
 
+--------------------------------------------------------------------------------
+-- | Loads a config without flattening any property object out.
+loadNotInlinedConfig :: (MonadIO m, MonadThrow m) => FilePath -> m Config
+loadNotInlinedConfig path
+    = do txt <- liftIO $ T.readFile path
+         pse <- parse path txt
+         let action
+                 = do ps         <- pse
+                      (tys, pts) <- typecheck $ rename ps
+                      return $ register tys pts
+         case action of
+             Left e    -> throwM e
+             Right reg -> return . Config . removeInlining . simplifyReg $ reg
+
+--------------------------------------------------------------------------------
+simplifyReg :: Reg -> Reg
+simplifyReg reg
+    = reg { regAST = fmap (simplify reg) (regAST reg) }
+
+--------------------------------------------------------------------------------
+removeInlining :: Reg -> Reg
+removeInlining reg
+    = reg { regAST = M.mapMaybeWithKey go (regAST reg) }
+  where
+    containsDot = T.any ('.' ==)
+
+    go key value
+        | containsDot key = Nothing
+        | otherwise       = Just $ unscoped key value
+
+    unscoped scope ast =
+        case astExpr ast of
+            OBJECT ps ->
+                let inner (Prop name child) =
+                        case T.stripPrefix (scope <> ".") name of
+                            Nothing ->
+                                Prop name (unscoped scope child)
+                            Just stripped ->
+                                let newScope = scope <> "." <> stripped in
```

Comment:
`scope <> "." <> stripped` is also `name`.  And in the `Nothing` case - where `scope` doesn't contain a dot - then `scope` == `name` too.  So basically this simplifies down to:

```haskell
                let inner (Prop name child) =
                        Prop stripped (unscoped name child) where
                            stripped = fromMaybe name $ T.stripPrefix (scope <> ".") name in
```

Reply:
It is not, because, before that patch, I kept the fully qualified name as the property name.

Reply:
I'm confused.  In my implementation & tests using `name` seems to do the right thing: https://github.com/dwijnand/deiko-config/tree/to-aeson-rebased.

Anyways, not too important, as this is a minor simplification.